src/Sulu/Bundle/PageBundle/Controller/IconController.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PageBundle\Controller;
+
+use FOS\RestBundle\View\ViewHandlerInterface;
+use HandcraftedInTheAlps\RestRoutingBundle\Routing\ClassResourceInterface;
+use Sulu\Component\Rest\AbstractRestController;
+use Sulu\Component\Rest\ListBuilder\CollectionRepresentation;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class IconController extends AbstractRestController implements ClassResourceInterface
+{
+    /**
+     * @var array<array<array{path: string, selection_json: string}>>
+     */
+    private array $iconSets;
+
+    /**
+     * @param array<array<array{path: string, selection_json: string}>> $iconSets
+     */
+    public function __construct(
+        ViewHandlerInterface $viewHandler,
+        array $iconSets,
+    ) {
+        parent::__construct($viewHandler);
+        $this->iconSets = $iconSets;
+    }
+
+    /**
+     * Returns icons.
+     *
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        $iconSet = (array) $this->iconSets[$request->query->get('icon_set')];
```

Comment:
would use `$this->getRequestParameter($request, 'icon_set', true);` to extract the name of the icon set to validate if the parameter is set